x11-wm/notion/files/notion-4.0.2-gcc14-build-fix.patch

```diff
@@ -0,0 +1,40 @@
+https://patch-diff.githubusercontent.com/raw/raboof/notion/pull/369.patch
+From: Brahmajit Das <brahmajit.xyz@gmail.com>
+Date: Wed, 1 May 2024 11:35:59 +0530
+Subject: [PATCH 1/1] Fix build with GCC 14 and above
+MIME-Version: 1.0
+Content-Type: text/plain; charset=UTF-8
+Content-Transfer-Encoding: 8bit
+
+Newer compilers such as GCC 14 ( and above ) have enabled a few compiler
+flags by default, -Wincompatible-pointer-types being one of them. Thus
+resulting in build errors such as:
+
+frame.c: In function ‘frame_set_numbers_extl’:
+frame.c:799:32: error: passing argument 1 of ‘ioncore_grab_establish’ from incompatible pointer type [-Wincompatible-pointer-types]
+  799 |         ioncore_grab_establish(frame, numbers_grab_handler, NULL,
+      |                                ^~~~~
+      |                                |
+      |                                WFrame * {aka struct WFrame_struct *}
+
+For now a type casting can be used to supress the error, as change the
+type of frame would require touching other parts of the codebase.
```

Comment:
For cases like this, please add a TODO in the code.

Also, on the upstream PR, I suggest making it very clear it's a workaround, and the proper fix would be to fix the frame type used, and you're open to suggestions on how to do it.

Reply:
I'll add a todo for now, also upstream did accept my patch and developer/s said "WFrame is a WMPlex which is a WWindow" so it should be fine.

https://github.com/raboof/notion/pull/369#pullrequestreview-2033171322